DatadogRUM/Sources/Instrumentation/Views/SwiftUI/SwiftUIViewNameExtractor.swift

```diff
@@ -0,0 +1,159 @@
+            return nil
+        }
+
+        // Reflector to inspect the view controller's internals
+        let reflector = createReflector(viewController)
+
+        return extractViewName(
+            from: viewController,
+            withReflector: reflector
+        )
+    }
+
+    private func extractViewName(
+        from viewController: UIViewController,
+        withReflector reflector: TopLevelReflector
+    ) -> String? {
+        let className = NSStringFromClass(type(of: viewController))
+        let controllerType = ControllerType(className: className)
+
+        switch controllerType {
+        case .tabItem:
+            return extractTabViewName(from: viewController)
+
+        case .hostingController:
+            if let output = SwiftUIViewPath.hostingController.traverse(with: reflector) {
+                return extractViewName(from: typeDescription(of: output))
+            }
+
+        case .navigationController:
+            // Try detail view first
+            if let output = SwiftUIViewPath.navigationStackDetail.traverse(with: reflector) {
+                return extractViewName(from: typeDescription(of: output))
+            }
+
+            // Check if it's a container view that we should ignore
+            if SwiftUIViewPath.navigationStackContainer.traverse(with: reflector) != nil {
+                return nil
+            }
+
+            // Try standard navigation stack view
+            if let output = SwiftUIViewPath.navigationStack.traverse(with: reflector) {
+                return extractViewName(from: typeDescription(of: output))
+            }
+
+        case .modal:
+            if let output = SwiftUIViewPath.sheetContent.traverse(with: reflector) {
+                return extractViewName(from: typeDescription(of: output))
+            }
+
+        case .unknown:
+            break
+        }
+
+        return nil
+    }
+
+    // MARK: - Helpers
+    /// Extracts a view name from a type description using regex patterns
+    internal func extractViewName(from input: String) -> String? {
+        // Pattern 1: Extract the view name from generic types like LazyView<ContentView>
+        let simpleGenericPattern = #"<([^<>,]+)>"#
+        if let match = input.range(of: simpleGenericPattern, options: .regularExpression) {
+            let startIndex = input.index(match.lowerBound, offsetBy: 1)
+            let endIndex = input.index(match.upperBound, offsetBy: -1)
+            return String(input[startIndex..<endIndex])
+        }
+
+        // Pattern 2: Extract the view name from complex generic types like ParameterizedLazyView<String, DetailView>
+        let complexGenericPattern = #"<.*,\s*([^<>,]+)>"#
+        if let match = input.range(of: complexGenericPattern, options: .regularExpression),
+           let captureRange = input.range(of: #"([^<>,]+)(?=>)"#, options: .regularExpression, range: match) {
+            return String(input[captureRange]).trimmingCharacters(in: .whitespaces)
+        }
```

Comment:
/suggestion You can use a single regex for this using the `NSRegularExpression`:
```swift
let regex = try NSRegularExpression(pattern: #"<(?:[^,>]*,\s+)?([^<>,]+?)>"#, options: [])
if
    let match = regex.firstMatch(in: input, options: [], range: NSRange(input.startIndex..<input.endIndex, in: input)),
    let range = Range(match.range(at: 1), in: input)
{
    return String(input[range])
}
```